app/src/org/commcare/dalvik/activities/LoginActivity.java

```diff
@@ -87,6 +110,7 @@
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         username.setInputType(InputType.TYPE_TEXT_FLAG_NO_SUGGESTIONS | InputType.TYPE_TEXT_VARIATION_VISIBLE_PASSWORD);
+        editTextColor = username.getCurrentTextColor();

```

Comment:
The other pattern would be having some state Enum that tracks whether we're in error mode or not